src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -213,24 +216,37 @@ internal static void AddArgumentsFromAppSettings(RpcWorkerDescription workerDesc
 
         internal bool ShouldAddWorkerConfig(string workerDescriptionLanguage)
         {
-            string workerRuntime = _environment.GetEnvironmentVariable(RpcWorkerConstants.FunctionWorkerRuntimeSettingName);
             if (_environment.IsPlaceholderModeEnabled())
             {
                 return true;
             }
 
-            if (!string.IsNullOrEmpty(workerRuntime))
+            if (!string.IsNullOrEmpty(_workerRuntime))
             {
-                _logger.LogDebug($"EnvironmentVariable {RpcWorkerConstants.FunctionWorkerRuntimeSettingName}: {workerRuntime}");
-                if (workerRuntime.Equals(workerDescriptionLanguage, StringComparison.OrdinalIgnoreCase))
+                _logger.LogDebug($"EnvironmentVariable {RpcWorkerConstants.FunctionWorkerRuntimeSettingName}: {_workerRuntime}");
+                if (_workerRuntime.Equals(workerDescriptionLanguage, StringComparison.OrdinalIgnoreCase))
                 {
                     return true;
                 }
                 // After specialization only create worker provider for the language set by FUNCTIONS_WORKER_RUNTIME env variable
-                _logger.LogInformation($"{RpcWorkerConstants.FunctionWorkerRuntimeSettingName} set to {workerRuntime}. Skipping WorkerConfig for language:{workerDescriptionLanguage}");
+                _logger.LogInformation($"{RpcWorkerConstants.FunctionWorkerRuntimeSettingName} set to {_workerRuntime}. Skipping WorkerConfig for language:{workerDescriptionLanguage}");
                 return false;
             }
             return true;
         }
+
+        internal void ReadLanguageWorkerFile(string workerPath)
+        {
+            if (_environment.IsPlaceholderModeEnabled() &&
+                 !string.IsNullOrEmpty(_workerRuntime) &&
```

Comment:
while we are doing this during specialization. why not do it for all language workers?

Reply:
done!